src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionTests.cs

```diff
@@ -32,6 +33,30 @@ void M()
             AssertJsonEquals(expected, results.First());
         }
 
+        [Fact]
+        public async Task TestGetCompletionsDoesNotIncludeUnimportedTypesAsync()
+        {
+            var markup =
+@"class A
+{
+    void M()
+    {
+        {|caret:|}
+    }
+}";
+            var (solution, locations) = CreateTestSolution(markup);
+
+            // Make sure the unimported types option is on by default.
+            solution = solution.WithOptions(solution.Options.WithChangedOption(CompletionServiceOptions.IsExpandedCompletion, true));
```

Comment:
Same here, please use `ShowItemsFromUnimportedNamespaces`. Also note that unless `IsExpandedCompletion` is true, the first time import completion is triggered we would not return any item. This is because we don't want to block completion if the cache for unimported items is not populated (first trigger would do that in background). So to test this, you can call `RunGetCompletionsAsync` twice to ensure nothing's returned.

Reply:
@genlu this doesn't work for some reason in the test.  I verified previously that the IsExpandedCompletion does make the completion list include unimported types, but this one does not.

I believe I would need to set both?  Since unimported types completion must be enabled and the list should include the expanded set?

Reply:
The provider filter looks like it only checks this IsExpandedCompletion - http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionServiceWithProviders.cs,157